auth/jwt_test.go

```diff
@@ -48,19 +41,120 @@ func (suite *JWTAuthenticatorTestSuite) createWrongToken(method jwt.SigningMetho
 	suite.Equal("Admin", user.Name)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestTokenHasClaims() {
+	token, err := GenerateToken("johndoe@example.org")
+	suite.Nil(err)
+	claims := jwt.MapClaims{}
+	_, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return []byte(config.GetString("auth.jwt.secret")), nil
+	})
+	suite.Nil(err)
+
+	userID, ok := claims["userid"]
+	suite.True(ok)
+	suite.Equal("johndoe@example.org", userID)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestTokenWithClaimsHasClaims() {
+	user := &TestUser{
+		Name:  "Admin",
+		Email: "johndoe@example.org",
+	}
+	database.GetConnection().Create(user)
+	token, err := GenerateTokenWithClaims(jwt.MapClaims{
+		"sub":    user.ID,
+		"userid": user.Email,
+	})
+	suite.Nil(err)
+	claims := jwt.MapClaims{}
+	_, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(config.GetString("auth.jwt.secret")), nil
+	})
+	suite.Nil(err)
+
+	userID, okID := claims["userid"]
+	suite.True(okID)
+	suite.Equal("johndoe@example.org", userID)
+	sub, okSub := claims["sub"]
+	suite.True(okSub)
+	suite.Equal(user.ID, uint(sub.(float64)))
+}
 
-	user = &TestUser{}
-	token, err = GenerateToken("wrongemail@example.org")
+func (suite *JWTAuthenticatorTestSuite) TestTokenWithClaimsValidates() {
+	token, err := GenerateToken("johndoe@example.org")
+	suite.Nil(err)
+	claims := jwt.MapClaims{}
+	_, err = jwt.ParseWithClaims(token, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(config.GetString("auth.jwt.secret")), nil
+	})
+	suite.Nil(err)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestAuthenticateWithClaims() {
+	tokenAuthenticator := &JWTAuthenticator{}
+	user := &TestUser{
+		Name:  "Admin",
+		Email: "johndoe@example.org",
+	}
+	database.GetConnection().Create(user)
+	token, err := GenerateTokenWithClaims(jwt.MapClaims{
+		"sub":    user.ID,
+		"userid": "johndoe@example.org",
+	})
+	suite.Nil(err)
+	suite.Nil(tokenAuthenticator.Authenticate(suite.createRequest(token), user))
+	suite.Equal("Admin", user.Name)
+}
+
+func (suite *JWTAuthenticatorTestSuite) TestGenerateTokenInvalidCredentials() {
+	tokenAuthenticator := &JWTAuthenticator{}
+	user := &TestUser{
+		Name:  "Admin",
+		Email: "johndoe@example.org",
+	}
+	database.GetConnection().Create(user)
+	token, err := GenerateToken("wrongemail@example.org")
 	suite.Nil(err)
 	suite.Equal("These credentials don't match our records.", tokenAuthenticator.Authenticate(suite.createRequest(token), user).Error())
+	suite.Nil(err)
+}
+func (suite *JWTAuthenticatorTestSuite) TestGenerateTokenWithClaimsInvalidCredentials() {
+	tokenAuthenticator := &JWTAuthenticator{}
+	user := &TestUser{
+		Name:  "Admin",
+		Email: "johndoe@example.org",
+	}
+	database.GetConnection().Create(user)
```

Comment:
```suggestion
	user := &TestUser{}
```
See my comment above (same reason)